pkg/util/iptables/iptables.go

```diff
@@ -783,3 +802,75 @@ func isResourceError(err error) bool {
 	}
 	return false
 }
+
+type ParseError interface {
+	Line() int
+	Error() error
+}
+
+type parseError struct {
+	cmd  string
+	line int
+}
+
+func (e parseError) Line() int {
+	return e.line
+}
+
+const (
+	colon = ":"
+)
+
+func (e parseError) Error() error {
+	return fmt.Errorf("%s: input error on line %d: ", e.cmd, e.line)
+}
+
+// parseRestoreError extracts the line from the error, if it matches returns parseError
+// for example:
+// input: iptables-restore: line 51 failed
+// output: parseError:  cmd = iptables-restore, line = 51
+// NOTE: parseRestoreError depends on the error format of iptables, if it ever changes
+// we need to update this function
+func parseRestoreError(err string) (ParseError, bool) {
+	errInfos := strings.SplitN(err, colon, 3)
```

Comment:
Why would this be 3?  The example above (and I confirmed) has 2 parts.

https://go.dev/play/p/69IpIPUBubN

Reply:
my mistake..😅